internal/ethapi/api.go

```diff
@@ -502,6 +502,53 @@ func (s *PublicBlockChainAPI) GetBalance(ctx context.Context, address common.Add
 	return (*hexutil.Big)(state.GetBalance(address)), state.Error()
 }
 
+// GetProof returns the Merkle-proof for a given account and optionally some storage keys.
+func (s *PublicBlockChainAPI) GetProof(ctx context.Context, address common.Address, storageKeys []string, blockNr rpc.BlockNumber) (map[string]interface{}, error) {
+	state, _, err := s.b.StateAndHeaderByNumber(ctx, blockNr)
+	if state == nil || err != nil {
+		return nil, err
+	}
+
+	storageTrie := state.StorageTrie(address)
+	storageHash := types.EmptyRootHash
+	codeHash := state.GetCodeHash(address)
+	storageProof := make([]map[string]interface{}, len(storageKeys))
+
+	if storageTrie != nil {
+		storageHash = storageTrie.Hash()
+	} else {
+		codeHash = crypto.Keccak256Hash(nil)
+	}
+
+	for i := range storageKeys {
```

Comment:
the common way to iterate over a list of things is 
```golang
for i, val := range storageKeys{
```

Reply:
thanks, changed it!